### PR TITLE
M2-4408: dataviz-useasuync-refactor

### PR DESCRIPTION
**Pull request by kdanilenka32**

### CODE REVIEW

**Review by nmialik — APPROVED** [review]

File: src/modules/Dashboard/features/RespondentData/RespondentDataReview/Feedback/FeedbackAssessment/FeedbackAssessment.tsx [hunk 1]
```diff
@@ -64,42 +60,46 @@ export const FeedbackAssessment = ({
...
+        assessmentItems: answers.map(({ answer, itemId }) => ({
+          itemId,
+          answers: answer.value,
+          edited: answer.edited,
+        })),
+      });
+      setActiveTab(FeedbackTabs.Reviewed);
+      setAssessmentStep(0);
+    } catch (error) {
+      console.warn(error);
```
Comment (IhorNesterenko99) [inline src/modules/Dashboard/features/RespondentData/RespondentDataReview/Feedback/FeedbackAssessment/FeedbackAssessment.tsx:101]: maybe show BE error in SubmitAssesmentPopup?

Comment (nmialik) [reply src/modules/Dashboard/features/RespondentData/RespondentDataReview/Feedback/FeedbackAssessment/FeedbackAssessment.tsx:101]: @IhorNesterenko99 this is not in the user stories, and this has not been implemented before. so I'd suggest to postpone this improvement for now, but inform BAs about it

File: src/modules/Dashboard/features/RespondentData/RespondentDataReview/Feedback/FeedbackReviewed/FeedbackReviewed.tsx [hunk 2]
```diff
@@ -19,35 +19,35 @@ export const FeedbackReviewed = () => {
...
+          userPublicKey: reviewerPublicKey,
+        } as Review;
+
+        return {
+          reviewer,
+          review: getDecryptedActivityData(encryptedData).decryptedAnswers,
+        };
+      });
+
+      setReviewers(decryptedData as ReviewData[]);
```
Comment (IhorNesterenko99) [inline src/modules/Dashboard/features/RespondentData/RespondentDataReview/Feedback/FeedbackReviewed/FeedbackReviewed.tsx]:
```suggestion
      setReviewers((decryptedData as ReviewData[]) ?? []);
```

File: src/modules/Dashboard/features/RespondentData/RespondentDataSummary/Report/Report.tsx [hunk 3]
```diff
@@ -79,32 +79,39 @@ export const Report = ({ activity, identifiers = [], versions = [] }: ReportProp
...
+            (contentDisposition &&
+              LATEST_REPORT_REGEX.exec(contentDisposition)?.groups?.filename) ??
+            LATEST_REPORT_DEFAULT_NAME;
+          const base64Str = Buffer.from(data).toString('base64');
+          const linkSource = getLatestReportUrl(base64Str);
+
+          download(linkSource, fileName, LATEST_REPORT_TYPE);
+        }
+      } catch (error) {
+        setLatestReportError(getErrorMessage(error));
```
Comment (IhorNesterenko99) [inline src/modules/Dashboard/features/RespondentData/RespondentDataSummary/Report/Report.tsx]: Do we need try catch here? Maybe just need to move setLatestReportError(getErrorMessage(error)) to the errorCallback of useAsync hook?

File: src/modules/Dashboard/features/RespondentData/RespondentDataSummary/Report/Report.tsx [hunk 4]
```diff
@@ -155,6 +162,8 @@ export const Report = ({ activity, identifiers = [], versions = [] }: ReportProp
 
         setSubscalesFrequency(subscalesFrequency);
         setResponseOptions(formattedResponses);
+      } catch (error) {
+        console.warn(error);
```
Comment (IhorNesterenko99) [inline src/modules/Dashboard/features/RespondentData/RespondentDataSummary/Report/Report.tsx:160]: Can we put an error message on the user interface to show that something went wrong in the BE side?

File: src/modules/Dashboard/features/RespondentData/RespondentDataSummary/Report/Subscales/AdditionalInformation/AdditionalInformation.tsx [hunk 5]
```diff
@@ -12,19 +12,18 @@ import { StyledHeader, StyledContent, StyledMdPreview } from './AdditionalInform
 
 export const AdditionalInformation = ({ optionText }: AdditionalInformationProps) => {
   const { t } = useTranslation();
-  const { execute: getOptionText } = useAsync(getOptionTextApi);
+  const { execute: getOptionText } = useAsync(getOptionTextApi, (response) =>
+    setAdditionalInformation(response?.data),
```
Comment (IhorNesterenko99) [inline src/modules/Dashboard/features/RespondentData/RespondentDataSummary/Report/Subscales/AdditionalInformation/AdditionalInformation.tsx]:
```suggestion
    setAdditionalInformation(response?.data ?? ''),
```

File: src/modules/Dashboard/features/RespondentData/RespondentDataSummary/RespondentDataSummary.tsx [hunk 6]
```diff
@@ -69,13 +65,15 @@ export const RespondentDataSummary = () => {
...
-        const versions = await getVersions({ appletId, activityId: selectedActivity.id });
+        const versions = await getVersionsApi({ appletId, activityId: selectedActivity.id });
         const versionsFilter = versions.data.result?.map(({ version }) => ({
           id: version,
           label: version,
         }));
         setValue('versions', versionsFilter);
         setVersions(versions.data.result);
+      } catch (error) {
+        console.warn(error);
```
Comment (IhorNesterenko99) [inline src/modules/Dashboard/features/RespondentData/RespondentDataSummary/RespondentDataSummary.tsx:76]: could you pls  show error message on UI?

File: src/modules/Dashboard/pages/RespondentData/RespondentData.tsx [hunk 7]
```diff
@@ -27,7 +27,9 @@ export const RespondentData = () => {
   const { ownerId } = workspaces.useData() || {};
   const respondentDataTabs = useRespondentDataTabs();
 
-  const { execute: getSummaryActivities } = useAsync(getSummaryActivitiesApi);
+  const { execute: getSummaryActivities } = useAsync(getSummaryActivitiesApi, (result) => {
+    setSummaryActivities(result?.data?.result);
```
Comment (IhorNesterenko99) [inline src/modules/Dashboard/pages/RespondentData/RespondentData.tsx]:
```suggestion
    setSummaryActivities(result?.data?.result ?? []);
```

File: src/modules/Dashboard/features/RespondentData/RespondentDataReview/Feedback/FeedbackAssessment/FeedbackAssessment.tsx (hunk 1 above)
Comment (IhorNesterenko99) [reply src/modules/Dashboard/features/RespondentData/RespondentDataReview/Feedback/FeedbackAssessment/FeedbackAssessment.tsx:101]: fine

File: src/modules/Dashboard/features/RespondentData/RespondentDataSummary/Report/Report.tsx (hunk 3 above)
Comment (kdanilenka32) [reply src/modules/Dashboard/features/RespondentData/RespondentDataSummary/Report/Report.tsx:101]: no we need here

**Review by IhorNesterenko99 — APPROVED** [review]

**Review by alex-i-e — APPROVED** [review]

**Pull request merged**